client.go

```diff
@@ -295,33 +304,25 @@ func (co *Conn) ReadMsgHeader(hdr *Header) ([]byte, error) {
 		err error
 	)
 
+	p = udpPool.Get().([]byte)
 	if isPacketConn(co.Conn) {
-		if co.UDPSize > MinMsgSize {
-			p = make([]byte, co.UDPSize)
-		} else {
-			p = make([]byte, MinMsgSize)
-		}
 		n, err = co.Read(p)
 	} else {
-		var length uint16
-		if err := binary.Read(co.Conn, binary.BigEndian, &length); err != nil {
-			return nil, err
-		}
-
-		p = make([]byte, length)
 		n, err = io.ReadFull(co.Conn, p)
```

Comment:
This is wrong. You're ignoring the length which can be up to 64KiB. This would limit the max message size under TCP to whatever the pool returns.